### PR TITLE
Remove V1 grpc API.

### LINKED ISSUE #1073: Remove V1 GRPC API
**Task description**

Completely remove the V1 GRPC API from the node.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Remove V1 API from the node.

Closes #1073 

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-consensus/src/Concordium/External.hs [hunk 1]
```diff
@@ -1000,381 +995,16 @@ toCString lbs = do
...
-
--- | Retrieve the last finalized block height relative to the most recent genesis index. Used for
--- resuming out-of-band catchup.
-getLastFinalizedBlockHeight ::
-    StablePtr ConsensusRunner ->
-    IO Word64
-getLastFinalizedBlockHeight cptr = do
-    (ConsensusRunner mvr) <- deRefStablePtr cptr
-    theBlockHeight <$> runMVR Q.getLastFinalizedBlockHeight mvr
-
```
Comment (td202) [inline concordium-consensus/src/Concordium/External.hs:1104]: It seems this function is still used in `ffi.rs`.

File: concordium-consensus/src/Concordium/External.hs [hunk 2]
```diff
@@ -1000,381 +995,16 @@ toCString lbs = do
...
--- | Get the list of transactions in a block with short summaries of their effects.
---  Returns a null-terminated string encoding a JSON value.
---  If the block hash is invalid or unknown, this returns the JSON null value.
---  For details of the value returned, see 'Concordium.Queries.Types.BlockSummary'.
-getBlockSummary :: StablePtr ConsensusRunner -> CString -> IO CString
-getBlockSummary cptr blockcstr =
-    decodeBlockHash blockcstr >>= \case
-        Nothing -> jsonCString AE.Null
-        Just bh -> jsonQuery cptr (Q.getBlockSummary bh)
```
Comment (td202) [inline concordium-consensus/src/Concordium/External.hs:1130]: Also remove `Queries.getBlockSummary`.

Comment (td202) [reply concordium-consensus/src/Concordium/External.hs:1130]: Also the `BlockSummary` type can (probably) be removed from base.

Comment (abizjak) [reply concordium-consensus/src/Concordium/External.hs:1130]: Yes.

File: concordium-consensus/src/Concordium/External.hs (hunk 1 above)
Comment (abizjak) [reply concordium-consensus/src/Concordium/External.hs]: Yes, indeed.

File: concordium-node/build.rs [hunk 3]
```diff
@@ -119,13 +116,6 @@ fn main() -> std::io::Result<()> {
...
     link_static_libs()?;
 
-    // build GRPC V1 interface.
-    tonic_build::configure()
-        .build_server(true)
-        .build_client(true) // the client is needed for the collector
-        .compile(&[&proto], &[&proto_root_input])
-        .expect("Failed to compile gRPC definitions!");
-
     build_grpc2(&proto_root_input)?;
```
Comment (EmilLa1) [inline concordium-node/build.rs:119]:
```suggestion
    build_grpc(&proto_root_input)?;
```
?

**Review by EmilLa1 — APPROVED** [review]
Looks good.

I think at some point we should rename all the "grpc2" stuff to simply being "grpc". There is now only one and only one grpc api of the node, so it could end up being a bit confusing the future having this "v2"..

File: concordium-node/build.rs (hunk 3 above)
Comment (abizjak) [reply concordium-node/build.rs:119]: No. I think that'll just cause more confusion.

**Comment by abizjak** [comment]
> I think at some point we should rename all the "grpc2" stuff to simply being "grpc". There is now only one and only one grpc api of the node, so it could end up being a bit confusing the future having this "v2"..

I am not convinced this is a good idea. It'll be more confusing and disruptive to do renaming. It's fairly easy to remember we used to have some other API.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased changes
 
+- Remove V1 GRPC API from the node. This removes configuration options
```
[the remark this answers is not in the document]
Comment (abizjak) [reply CHANGELOG.md:5]: I don't think mentioning V2 is necessary.

**Review by td202 — APPROVED** [review]

**Pull request merged**